plugins/kubernetes/app/assets/javascripts/kubernetes/kubernetes.js

```diff
@@ -5,31 +5,57 @@ angular.module('samson').config(function($stateProvider, $urlRouterProvider) {
   $stateProvider
     .state('kubernetes', {
       url: '/projects/:project_id/kubernetes',
-      template: '<ui-view/>',
       abstract: true
     })
+
+    /*
+       Kubernetes Roles
+     */
     .state('kubernetes.roles', {
       url: "/roles",
       data: {
         'selectedTab': 0
       },
       views: {
-        'roles@': {
+        'content@': {
           templateUrl: 'kubernetes/kubernetes_roles.tmpl.html',
           controller: 'KubernetesRolesCtrl'
         }
       }
     })
+    .state('kubernetes.roles.edit', {
+      url: "/:role_id/edit",
+      views: {
+        'content@': {
+          templateUrl: 'kubernetes/kubernetes_edit_role.tmpl.html',
+          controller: 'KubernetesEditRoleCtrl'
+        }
+      }
+    })
+    .state('kubernetes.roles.create', {
+      url: "/new",
```

Comment:
This url is relative to the parent state, which is 'kubernetes.roles'. So, when you navigate to this state, the url, in the browser, will actually be '/roles/new'.